aten/src/ATen/native/cuda/Blas.cu

```diff
@@ -6,26 +6,27 @@ namespace at { namespace native {
 
 Tensor &addmv_impl_cuda(Tensor& result, const Tensor &self, const Tensor &mat, const Tensor &vec, Scalar beta_, Scalar alpha_) {
   auto r_stride = result.stride(0);
-  auto vec_stride = vec.stride(0);
+  const auto vec_contiguous = vec.contiguous();
```

Comment:
You are unconditionally calling vec.contiguous(), whereas you only need to do it if vec_stride of the original vector is equal to 0.

Reply:
Sure. Have updated.

I actually assumed that the unconditional `contiguous` wouldn't be an overhead for already contiguous `vec` and that it would be similar to checking `vec_stride == 0` and the branching on it.

Reply:
It won't, but there are many non-contiguous Tensors that don't have stride=0 and that are supported.

Reply:
Oh, I understand now. Thank You :).